auth_service/user_management/claims_repository/router.py

```diff
@@ -244,3 +246,48 @@ async def delete_claim(
         raise claim_not_found_error from error
 
     return Response(status_code=204)
+
+
+@router.get(
+    "/datasets/{dataset_id}/users/{user_id}/download-access",
+    operation_id="get_download_access",
+    tags=["datasets"],
+    summary="Check download access permission",
+    description="Endpoint to check whether the given dataset"
+    " can be downloaded by the given user. For internal use only.",
+    responses={
+        200: {"model": list[Claim], "description": "User claims have been retrieved."},
+        404: {"description": "The user was not found."},
+        422: {"description": "Validation error in submitted user IDs."},
+    },
+    status_code=200,
+)
+async def get_download_access(
+    dataset_id: str = Path(
+        ...,
+        alias="dataset_id",
+        title="Internal ID of the user",
+    ),
+    user_id: str = Path(
+        ...,
+        alias="user_id",
+        title="Internal ID of the user",
+    ),
+    user_dao: UserDao = Depends(get_user_dao),
+    claim_dao: ClaimDao = Depends(get_claim_dao),
+    # internal service, authorization without token via service mesh
+) -> Literal["false", "true"]:
+    """Check download access permission"""
+    if not await user_exists(user_id, user_dao):
+        raise user_not_found_error
+
+    async for claim in claim_dao.find_all(
+        mapping={
+            "user_id": user_id,
+            "visa_type": VisaType.CONTROLLED_ACCESS_GRANTS,
+        }
+    ):
+        if is_valid_claim(claim) and has_download_access_for_dataset(claim, dataset_id):
+            return "true"
```

Comment:
PMO: should go to the core (is already in refactoring list)

Reply:
The whole auth service will be refactored, using dependency injection properly and split into separate services anyway. For now I think this is good enough.